doc/sphinx/language/gallina-extensions.rst

```diff
@@ -869,255 +870,223 @@ Module system
+.. _terminating_module:
 
-       Defines a functor with parameters given by the list of :token:`module_binding` and body :token:`module_expression`.
+**Terminating an interactive module or module type definition**
 
-    .. cmdv:: Module @ident {* @module_binding} : @module_type := @module_expression
+Interactive modules are terminated with the :cmd:`End` command, which
+is also used to terminate :ref:`Sections<section-mechanism>`.
+:n:`End @ident` closes the interactive module or module type :token:`ident`.
+If the module type was given, the command verifies that the content of the module
+matches the module type.  If the module is not a
+functor, its components (constants, inductive types, submodules etc.)
+are now available through the dot notation.
 
-       Defines a functor with parameters given by the list of :token:`module_binding` (possibly none), and output module type :token:`module_type`,
-       with body :token:`module_expression`.
+.. exn:: No such label @ident.
+    :undocumented:
 
-    .. cmdv:: Module @ident {* @module_binding} <: {+<: @module_type} := @module_expression
-
-       Defines a functor with parameters given by module_bindings (possibly none) with body :token:`module_expression`.
-       The body is checked against each :n:`@module_type__i`.
-
-    .. cmdv:: Module @ident {* @module_binding} := {+<+ @module_expression}
-
-       is equivalent to an interactive module where each :token:`module_expression` is included.
-
-.. cmd:: Module Type @ident
-
-   This command is used to start an interactive module type :token:`ident`.
-
-   .. cmdv:: Module Type @ident {* @module_binding}
-
-      Starts an interactive functor type with parameters given by :token:`module_bindings`.
+.. exn:: Signature components for label @ident do not match.
+    :undocumented:
 
+.. exn:: The field @ident is missing in @qualid.
+   :undocumented:
 
-Reserved commands inside an interactive module type:
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. |br| raw:: html
 
-.. cmd:: Include @module
+    <br>
 
-   Same as ``Include`` inside a module.
+.. note::
 
-.. cmd:: Include {+<+ @module}
+  #. Interactive modules and module types can be nested.
+  #. Interactive modules and module types can't be defined inside of :ref:`sections<section-mechanism>`.
+     Sections can be defined inside of interactive modules and module types.
+  #. Hints and notations (:cmd:`Hint` and :cmd:`Notation` commands) can also appear inside interactive
+     modules and module types. Note that with module definitions like:
 
-   This is a shortcut for the command :n:`Include @module` for each :token:`module`.
+     :n:`Module @ident__1 : @module_type := @ident__2.`
 
-.. cmd:: @assumption_token Inline @assums
-   :name: Inline
+     or
 
-   The instance of this assumption will be automatically expanded at functor application, except when
-   this functor application is prefixed by a ``!`` annotation.
```

Comment:
This is what I'm talking about.

Reply:
I think I got it.

You know I'm working primarily from the Conversation, not from the Files Changed.  It's not easy to get the context for a lot of your comments.  All I see is this (below), so it's not at all apparent that you're talking about the Inline clause of assumptions:

![image](https://user-images.githubusercontent.com/1253341/77366741-e3189000-6d15-11ea-9b28-66bdb71052c7.png)